internal/fs/inode/file_test.go

```diff
@@ -676,6 +676,180 @@ func (t *FileTest) TestTestTruncateDownwardForLocalFileShouldUpdateLocalFileAttr
 	assert.Equal(t.T(), "gcs.NotFoundError: object test not found", err.Error())
 }
 
+func (t *FileTest) TestTruncateUpwardForLocalFileWhenStreamingWritesAreEnabled() {
+	tbl := []struct {
+		name         string
+		performWrite bool
+	}{
+		{
+			name:         "WithWrite",
+			performWrite: true,
+		},
+		{
+			name:         "WithOutWrite",
+			performWrite: false,
+		},
+	}
+	for _, tc := range tbl {
+		t.Run(tc.name, func() {
+			// Create a local file inode.
+			t.createInodeWithLocalParam("test", true)
+			t.in.writeConfig = getWriteConfig()
+			err := t.in.CreateBufferedOrTempWriter()
+			assert.Nil(t.T(), err)
+			assert.NotNil(t.T(), t.in.bwh)
+
+			// Fetch the attributes and check if the file is empty.
+			attrs, err := t.in.Attributes(t.ctx)
+			assert.Nil(t.T(), err)
+			assert.Equal(t.T(), uint64(0), attrs.Size)
+
+			if tc.performWrite {
+				err := t.in.Write(t.ctx, []byte("hi"), 0)
+				assert.Nil(t.T(), err)
+				assert.Equal(t.T(), int64(2), t.in.bwh.WriteFileInfo().TotalSize)
+				// Fetch the attributes and check if the file size reflects the write.
+				attrs, err := t.in.Attributes(t.ctx)
+				assert.Nil(t.T(), err)
+				assert.Equal(t.T(), uint64(2), attrs.Size)
+			}
+
+			err = t.in.Truncate(t.ctx, 10)
+
+			assert.Nil(t.T(), err)
+			// The inode should return the new size.
+			attrs, err = t.in.Attributes(t.ctx)
+			assert.Nil(t.T(), err)
+			assert.Equal(t.T(), uint64(10), attrs.Size)
+			// Data shouldn't be updated to GCS.
+			statReq := &gcs.StatObjectRequest{Name: t.in.Name().GcsObjectName()}
+			_, _, err = t.bucket.StatObject(t.ctx, statReq)
+			assert.NotNil(t.T(), err)
+			assert.Equal(t.T(), "gcs.NotFoundError: object test not found", err.Error())
+		})
+	}
+}
+
+func (t *FileTest) TestTruncateUpwardForEmptyGCSFileWhenStreamingWritesAreEnabled() {
```

Comment:
Nit: we can merge this with TestTruncateUpwardForLocalFileWhenStreamingWritesAreEnabled